processing/src/main/java/org/apache/druid/segment/data/ImmutableRTreeObjectStrategy.java

```diff
@@ -63,8 +63,9 @@ public Class<ImmutableRTree> getClazz()
   @Override
   public ImmutableRTree fromByteBuffer(ByteBuffer buffer, int numBytes)
   {
-    buffer.limit(buffer.position() + numBytes);
-    return new ImmutableRTree(buffer, bitmapFactory);
+    ByteBuffer duplicateBuf = buffer.duplicate();
```

Comment:
Please add a comment explaining the need for a duplicate buffer for future reference.

Reply:
fixed